micrometer-spring-legacy/src/main/java/io/micrometer/spring/advice/TimedAnnotationAdvisor.java

```diff
@@ -0,0 +1,52 @@
+package io.micrometer.spring.advice;
+
+import io.micrometer.core.annotation.Timed;
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.springframework.aop.Pointcut;
+import org.springframework.aop.support.AbstractPointcutAdvisor;
+import org.springframework.aop.support.ComposablePointcut;
+import org.springframework.aop.support.annotation.AnnotationMatchingPointcut;
+import org.springframework.aop.support.annotation.AnnotationMethodMatcher;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+public class TimedAnnotationAdvisor extends AbstractPointcutAdvisor {
+
+    private final Pointcut pointcut;
+    private final MethodInterceptor advice;
+
+    public TimedAnnotationAdvisor(TimedMethodInterceptor timedMethodInterceptor) {
+        Pointcut cpc = new AnnotationMatchingPointcut(Timed.class, true);
+        Pointcut mpc = new ComposablePointcut(new InheritedAnnotationMethodMatcher(Timed.class));
```

Comment:
` InheritedAnnotationMethodMatcher` looks for `@Timed` annotation not only on implementation method, but also goes to interface method. In Spring Framework 5.0 it can be replaced by `new AnnotationMatchingPointcut(null, Timed.class, true)`.